logstash-core/spec/logstash/java_pipeline_spec.rb

```diff
@@ -226,6 +237,38 @@ def flush(options)
     end
   end
 
+  context "a crashing worker" do
+    subject { mock_java_pipeline_from_string(config, pipeline_settings_obj) }
+
+    let(:pipeline_settings) { { "pipeline.batch.size" => 1, "pipeline.workers" => 1 } }
+    let(:config) do
+      <<-EOS
+      input { generator {} }
+      filter { dummycrashingfilter {} }
+      output { dummyoutput {} }
+      EOS
+    end
+    let(:dummyoutput) { ::LogStash::Outputs::DummyOutput.new }
+
+    before :each do
+      allow(::LogStash::Outputs::DummyOutput).to receive(:new).with(any_args).and_return(dummyoutput)
+      allow(LogStash::Plugin).to receive(:lookup).with("input", "generator").and_return(LogStash::Inputs::Generator)
+      allow(LogStash::Plugin).to receive(:lookup).with("codec", "plain").and_return(LogStash::Codecs::Plain)
+      allow(LogStash::Plugin).to receive(:lookup).with("filter", "dummycrashingfilter").and_return(DummyCrashingFilter)
+      allow(LogStash::Plugin).to receive(:lookup).with("output", "dummyoutput").and_return(::LogStash::Outputs::DummyOutput)
+    end
+
+    after :each do
+      subject.shutdown
+    end
+
+    it "does not raise in the main thread, terminates the run thread and finishes execution" do
+      subject.start
```

Comment:
since the issue was that before the exception would bubble up,  maybe we should define the test in the negative. for example:

```ruby
it "does not raise in the main thread" do
  expect { subject.start && subject.shutdown }.to_not raise_error
end
```

Reply:
added to your suggestion.